app/src/main/java/org/mozilla/focus/experiments/NimbusSetup.kt

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.experiments
+
+import android.content.Context
+import android.net.Uri
+import mozilla.components.service.nimbus.NimbusApi
+import mozilla.components.service.nimbus.Nimbus
+import mozilla.components.service.nimbus.NimbusAppInfo
+import mozilla.components.service.nimbus.NimbusDisabled
+import mozilla.components.service.nimbus.NimbusServerSettings
+import mozilla.components.support.base.log.logger.Logger
+import org.mozilla.focus.BuildConfig
+import org.mozilla.focus.R
+import org.mozilla.focus.ext.components
+import org.mozilla.focus.telemetry.TelemetryWrapper
+
+@Suppress("TooGenericExceptionCaught")
+fun createNimbus(context: Context, url: String?): NimbusApi =
+    try {
+        // Eventually we'll want to use `NimbusDisabled` when we have no NIMBUS_ENDPOINT.
+        // but we keep this here to not mix feature flags and how we configure Nimbus.
+        val serverSettings = if (!url.isNullOrBlank()) {
+            NimbusServerSettings(url = Uri.parse(url))
+        } else {
+            null
+        }
+
+        // Global opt out state is stored in Nimbus, and shouldn't be toggled to `true`
+        // from the app unless the user does so from a UI control.
+        // However, the user may have opt-ed out of making experiments already, so
+        // we should respect that setting here.
+        // TODO change this to separate pref if we need a separate UI control.
+        val enabled = TelemetryWrapper.isTelemetryEnabled(context)
+
+        // The name "focus-android" or "klar-android" here corresponds to the app_name defined
+        // for the family of apps that encompasses all of the channels for the Focus app.
+        // This is defined upstream in the telemetry system. For more context on where the
+        // app_name come from see:
+        // https://probeinfo.telemetry.mozilla.org/v2/glean/app-listings
+        // and
+        // https://github.com/mozilla/probe-scraper/blob/master/repositories.yaml
+        val appInfo = NimbusAppInfo(
+            appName = getNimbusAppName(),
+            // Note: Using BuildConfig.BUILD_TYPE is important here so that it matches the value
+            // passed into Glean. `Config.channel.toString()` turned out to be non-deterministic
+            // and would mostly produce the value `Beta` and rarely would produce `beta`.
+            channel = BuildConfig.BUILD_TYPE
+        )
+        Nimbus(context, appInfo, serverSettings).apply {
```

Comment:
I'd like to see an error reporter passed here: e.g. https://github.com/mozilla-mobile/fenix/pull/20049/files#diff-1e627319338ce89fe0c17b48d5e02d49fcf559ff01e8400a623bbbca1c6f9e73R24-R29

Reply:
We do the same logic in the catch below already by forwarding the exceptions to the crash reporter. Is there something else we're looking to add?

Reply:
Ping @jhugman